shell/platform/darwin/macos/framework/Source/FlutterMouseCursorPlugin.mm

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <objc/message.h>
+
+#import "FlutterMouseCursorPlugin.h"
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterCodecs.h"
+
+static NSString* const kMouseCursorChannel = @"flutter/mousecursor";
+
+static NSString* const kActivateSystemCursorMethod = @"activateSystemCursor";
+static NSString* const kKindKey = @"kind";
+static NSString* const kDeviceKey = @"device";
+
+static NSString* const kKindValueNone = @"none";
+
+static NSMutableDictionary* cachedSystemCursors;
```

Comment:
This should be an ivar, not global, so that it doesn't outlive the plugin in an add-to-app scenario.